app/workflow-controller.go

```diff
@@ -102,29 +102,21 @@ func NewWorkflowControllerApp(c *Config) *WorkflowController {
 		}
 	}
 
-	workflowClient, err := wclient.NewWorkflowClient(kubeConfig)
+	client, err := wclient.NewWorkflowClient(kubeConfig)
 	if err != nil {
-		glog.Fatalf("Unable to initialize a Workflow client:%v", err)
+		glog.Fatalf("Unable to initialize a client:%v", err)
 	}
 
 	kubeInformerFactory := kubeinformers.NewSharedInformerFactory(kubeClient, time.Second*30)
 
-	workflowInformerFactory := winformers.NewSharedInformerFactory(workflowClient, time.Second*30)
-	workflowCtrl := controller.NewWorkflowController(workflowClient, kubeClient, kubeInformerFactory, workflowInformerFactory)
+	workflowInformerFactory := winformers.NewSharedInformerFactory(client, time.Second*30)
```

Comment:
These factories (here and in the following lines, all the New*() calls) can never fail? i.e they return no error and never return nil?

Reply:
Basically no. They simply allocate the resource. And golang don't throw in ctor ;)